presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -546,6 +556,52 @@ private RelationPlan planLateralJoin(Join join, RelationPlan leftPlan, Lateral l
         return new RelationPlan(planBuilder.getRoot(), analysis.getScope(join), outputSymbols);
     }
 
+    private TranslationMap filterExpressionTranslationMap(RelationPlan leftPlan, RelationPlan rightPlan, PlanBuilder leftPlanBuilder, PlanBuilder rightPlanBuilder, Join join)
```

Comment:
I don't see much value in having this as a separate method. There's no reuse across various operations and it's tied to the specific implementation of planning join + lateral. I would inline it in the method that uses is and add some comments in the appropriate sections if there's any logical separation worth highlighting.

Reply:
I thought it would be better to graphically separate it, because it constructs the `JoinNode` which is *not* a part of the resulting plan (as mentioned above). Inlining the construction could be misleading at the first view.
Surely inlining + comment would be OK.